open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +3786,173 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+
+    PlanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+        plan-task:
+          $ref: '#/components/schemas/PlanTask'
+        stats-fields:
+          description:
+            A list of fields that the client requests the server to send statistics
+            in each `FileScanTask` returned in the response
+          type: array
+          items:
+            $ref: '#/components/schemas/FieldName'
+
+    TableScanContext:
+      anyOf:
```

Comment:
Does this need to be `oneOf` instead of `anyOf`? https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/

We can have either a snapshot scan or an incremental scan but we should never have both in the request.

Reply:
I was mainly following how the `TableUpdate` and `BaseUpdate` are written right now:

```yaml
    TableUpdate:
      anyOf:
        - $ref: '#/components/schemas/AssignUUIDUpdate'
        - $ref: '#/components/schemas/UpgradeFormatVersionUpdate'
        - $ref: '#/components/schemas/AddSchemaUpdate'
...
```

```yaml
    BaseUpdate:
      discriminator:
        propertyName: action
        mapping:
          assign-uuid: '#/components/schemas/AssignUUIDUpdate'
          upgrade-format-version: '#/components/schemas/UpgradeFormatVersionUpdate'
...
```

Is that a difference case from the one here?

Reply:
From the link in @amogh-jahagirdar's first comment it looks like `oneOf` is more correct because `anyOf` would allow mixing the different options together. We should probably answer the question of what is correct when using a descriminator as well and follow up with a fix for that structure. But since there is no descriminator here, I think it should be `oneOf`.

Reply:
@jackye1995 @amogh-jahagirdar @rdblue I think i agree that it should probably just be `oneOf` as thats more clear for a spec reader to know its either a `SnapshotScanContext` or a `IncrementalSnapshotScanContext`.

However from the link that amogh shared https://swagger.io/docs/specification/data-models/inheritance-and- polymorphism/ the doc states that a spec writer can use either `oneOf` or `anyOf`, when using `discriminator`.

>Polymorphism
 In your API, you can have request and responses that can be described by several alternative schemas. In OpenAPI 3.0, to describe such a model, you can use the oneOf or anyOf keywords

>In our example, the discriminator points to the objectType property that contains the data type name. The discriminator is used with anyOf or oneOf keywords only. It is important that all the models mentioned below anyOf or oneOf contain the property that the discriminator specifies. This means, for example, that in our code above, both simpleObject and complexObject must have the objectType property

For now though will change this to be `oneOf`.